java-client/src/test/java/org/opensearch/client/transport/util/GeneratedCertificateSSLContext.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.transport.util;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.security.KeyManagementException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.SecureRandom;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Random;
+import java.util.Vector;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import javax.net.ssl.SSLContext;
+import org.apache.hc.core5.ssl.SSLContextBuilder;
+import org.opensearch.client.util.TriConsumer;
+import org.opensearch.client.util.TriFunction;
+import sun.security.util.KnownOIDs;
+import sun.security.util.ObjectIdentifier;
+import sun.security.x509.AlgorithmId;
+import sun.security.x509.AuthorityKeyIdentifierExtension;
+import sun.security.x509.BasicConstraintsExtension;
+import sun.security.x509.CertificateAlgorithmId;
+import sun.security.x509.CertificateExtensions;
+import sun.security.x509.CertificateSerialNumber;
+import sun.security.x509.CertificateValidity;
+import sun.security.x509.CertificateVersion;
+import sun.security.x509.CertificateX509Key;
+import sun.security.x509.DNSName;
+import sun.security.x509.ExtendedKeyUsageExtension;
+import sun.security.x509.Extension;
+import sun.security.x509.GeneralName;
+import sun.security.x509.GeneralNames;
+import sun.security.x509.IPAddressName;
+import sun.security.x509.KeyIdentifier;
+import sun.security.x509.KeyUsageExtension;
+import sun.security.x509.SerialNumber;
+import sun.security.x509.SubjectAlternativeNameExtension;
+import sun.security.x509.SubjectKeyIdentifierExtension;
+import sun.security.x509.X500Name;
+import sun.security.x509.X509CertImpl;
+import sun.security.x509.X509CertInfo;
+
+@SuppressWarnings("sunapi")
+public class GeneratedCertificateSSLContext {
```

Comment:
I think it would be much simpler to pre-generate certs, no? (see please https://github.com/opensearch-project/OpenSearch/tree/main/plugins/transport-reactor-netty4/src/test/resources).

Reply:
(at worst, the security plugin uses Bouncycastle to generate those https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/ssl/CertificatesRule.java)

Reply:
Pre-generating and committing the certs unfortunately leads to a lot of false positive security alerts. I'd initially avoided using BC seeing as the changes were with the aim to avoid pulling in another dependency. But can swap over to BC